container-host-files/etc/scf/config/scripts/forward_logfiles.sh

```diff
@@ -1,175 +1,171 @@
+
+# used for initial config
+CONF_OFFSET="10" # Before 50-default.conf to avoid re-logging our output
+MAIN_CONFIG="${CONF_OFFSET}-vcap.conf"
 RSYSLOG_CONF_DIR="/etc/rsyslog.d"
+PID_FILE="/var/run/rsyslogd.monit.pid"
 
-#used for adding individual logs
-BACKUP_WATCH_DIR=/var/vcap/sys/log #in case no ENV variable is set for RSYSLOG_FORWARDER_WATCH_DIR
-RSYSLOG_CONF_PREFIX=91-vcap
-RSYSLOG_CONF_DIR=/etc/rsyslog.d
+# used for adding individual logs
+BACKUP_WATCH_DIR="/var/vcap/sys/log" #in case no ENV variable is set for RSYSLOG_FORWARDER_WATCH_DIR
+RSYSLOG_CONF_PREFIX="$((CONF_OFFSET + 1))-vcap"
 IGNORE_DIR="gocode"
 TARGET_NAME=
 TARGET_BASENAME=
 
-SCRIPT_FILE=/usr/sbin/forward_logfiles.sh
-PB_OUT=/var/log/pb.out
+SCRIPT_FILE="/usr/sbin/forward_logfiles.sh"
+PB_OUT="/var/log/pb.out"
 
-if [ ! -f $SCRIPT_FILE ]; then
-	echo "#log forwarding script" > $PB_OUT
+if [ ! -f "${SCRIPT_FILE}" ]; then
+	echo "#log forwarding script" > "${PB_OUT}"
 fi
 
 if [ -z "${RSYSLOG_FORWARDER_WATCH_DIR:-}" ]; then
-        RSYSLOG_FORWARDER_WATCH_DIR=$BACKUP_WATCH_DIR
-        echo "RSYSLOG_FORWARDER_WATCH_DIR not set. Using default $BACKUP_WATCH_DIR" >> $PB_OUT
+        RSYSLOG_FORWARDER_WATCH_DIR="${BACKUP_WATCH_DIR}"
+        echo "RSYSLOG_FORWARDER_WATCH_DIR not set. Using default ${BACKUP_WATCH_DIR}" >> "${PB_OUT}"
 fi
 
-if [ ! -d "$RSYSLOG_FORWARDER_WATCH_DIR" ]; then
-        echo "$RSYSLOG_FORWARDER_WATCH_DIR is not a valid directory" >> $PB_OUT
+if [ ! -d "${RSYSLOG_FORWARDER_WATCH_DIR}" ]; then
+        echo "${RSYSLOG_FORWARDER_WATCH_DIR} is not a valid directory" >> "${PB_OUT}"
 fi
 
-if [ -z "$SCF_LOG_HOST" ]; then
-    echo "SCF_LOG_HOST is not set" >> $PB_OUT
+if [ -z "${SCF_LOG_HOST}" ]; then
+    echo "SCF_LOG_HOST is not set" >> "${PB_OUT}"
     exit 0
 fi
 
-_SCF_LOG_CONFIG=$SCF_LOG_HOST:$SCF_LOG_PORT
-
 function appendToCron {
-	echo appending to cron
+        echo appending to cron
         crontab -l > tempcrontab
-        echo "export _SCF_LOG_CONFIG=$SCF_LOG_CONFIG" >> $SCRIPT_FILE
-        echo "export RSYSLOG_FORWARDER_WATCH_DIR=$RSYSLOG_FORWARDER_WATCH_DIR" >> $SCRIPT_FILE
-        echo "export SCF_LOG_HOST=$SCF_LOG_HOST" >> $SCRIPT_FILE
-        echo "export SCF_LOG_PORT=$SCF_LOG_PORT" >> $SCRIPT_FILE
-        echo "export SCF_LOG_PREFIX=$SCF_LOG_PREFIX" >> $SCRIPT_FILE
-        echo "export SCF_LOG_PROTOCOL=$SCF_LOG_PROTOCOL" >> $SCRIPT_FILE
-        cat $0 >> $SCRIPT_FILE
-        echo "*/1 * * * * bash $SCRIPT_FILE >> /dev/null 2>&1" >> tempcrontab
+        sed 's@^\s*@@' >>"${SCRIPT_FILE}" <<-EOF
+                export RSYSLOG_FORWARDER_WATCH_DIR=${RSYSLOG_FORWARDER_WATCH_DIR}
+                export SCF_LOG_HOST=${SCF_LOG_HOST}
+                export SCF_LOG_PORT=${SCF_LOG_PORT}
+                export SCF_LOG_PREFIX=${SCF_LOG_PREFIX}
+                export SCF_LOG_PROTOCOL=${SCF_LOG_PROTOCOL}
+	EOF
+        cat "$0" >> "${SCRIPT_FILE}"
+        echo "*/1 * * * * /usr/bin/env bash ${SCRIPT_FILE} >> /dev/null 2>&1" >> tempcrontab
```

Comment:
The whole "stick ourselves in cron to find new files" thing is kinda ugly. Wish we could rewrite this to use inotify instead.

Reply:
Mini simplification opp: `*/1` means "every minute in steps of 1 minute".
That is the same as `*`, i.e. "every minute".